torchci/pages/nightlies.tsx

```diff
@@ -0,0 +1,453 @@
+    tooltip: {
+      trigger: "axis",
+      formatter: (params: any) => {
+        const red = params[0].data.red;
+        const green = params[0].data.green;
+        const pending = params[0].data.pending;
+        const total = params[0].data.total;
+
+        const redPct = ((red / total) * 100).toFixed(2) + "%";
+        const greenPct = ((green / total) * 100).toFixed(2) + "%";
+        const pendingPct = ((pending / total) * 100).toFixed(2) + "%";
+        return `Red: ${red} (${redPct})<br/>Green: ${green} (${greenPct})<br/>Pending: ${pending} (${pendingPct})<br/>Total: ${total}`;
+      },
+    },
+  };
+
+  return (
+    <Paper sx={{ p: 2, height: "100%" }} elevation={3}>
+      <ReactECharts
+        style={{ height: "100%", width: "100%" }}
+        option={options}
+      />
+    </Paper>
+  );
+}
+
+// Specialized version of TablePanel for TTS metrics.
+function TTSPanel({
+  title,
+  queryName,
+  queryParams,
+  metricHeaderName,
+  metricName,
+  branchName,
+}: {
+  title: string;
+  queryName: string;
+  queryParams: RocksetParam[];
+  metricHeaderName: string;
+  metricName: string;
+  branchName: string;
+}) {
+  return (
+    <TablePanel
+      title={title}
+      queryName={queryName}
+      queryParams={queryParams}
+      columns={[
+        {
+          field: metricName,
+          headerName: metricHeaderName,
+          flex: 1,
+          valueFormatter: (params: GridValueFormatterParams<number>) =>
+            durationDisplay(params.value),
+        },
+        { field: "count", headerName: "Count", flex: 1 },
+        {
+          field: "name",
+          headerName: "Name",
+          flex: 5,
+          // valueFormatter only treat the return value as string, so we need
+          // to use renderCell here to get the JSX
+          renderCell: (params: GridRenderCellParams<string>) => {
+            const jobName = params.value;
+            if (jobName === undefined) {
+              return `Invalid job name ${jobName}`;
+            }
+
+            const encodedJobName = encodeURIComponent(jobName);
+            const encodedBranchName = encodeURIComponent(branchName);
+            return (
+              <a
+                href={`/tts/pytorch/pytorch/${encodedBranchName}?jobName=${encodedJobName}`}
+              >
+                {jobName}
+              </a>
+            );
+          },
+        },
+      ]}
+      dataGridProps={{ getRowId: (el: any) => el.name }}
+    />
+  );
+}
+
+function TimePicker({ label, value, setValue }: any) {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <DateTimePicker
+        renderInput={(props) => <TextField {...props} />}
+        label={label}
+        value={value}
+        onChange={(newValue) => {
+          setValue(newValue);
+        }}
+      />
+    </LocalizationProvider>
+  );
+}
+
+/**
+ * Allows the user to pick from common time ranges, or manually set their own.
+ */
+export function TimeRangePicker({
```

Comment:
Also you don't need to copy this function and could import it from `metrics.tsx` I think